### PR TITLE
Added AvroCompatibilityHelper(https://github.com/linkedin/avro-util) support on Hive 1.0.1 to help hive consumers upgrade to higher Avro version

### PR DESCRIPTION
**Pull request by abhisheknath2011**
### What changes were proposed in this pull request?
1. Usage of AvroCompatibilityHelper to replace deprecated Avro methods.
2. AvroCompatibilityHelper to help hive consumer easily migrate to the higher version of Avro.
3. Added https://linkedin.jfrog.io repository to pull avro-util dependency.
3. No need to upgrading from Jackson 1.x to Jackson 2.x as compatibility helper handles internally.
4. Changes are done on a branch created from branch-1.0.1.

### Why are the changes needed?
1. Gobblin currently uses hive 1.0.1 and the upgrade of Avro 1.9.2 is blocked due older version of Avro and Jackson on hive 1.0.1.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
1. Local build
2. Released a local snapshot and applied on Gobblin.
3. With these changes gobblin build failures coming from hive serde library is resolved and gobblin is able to use Avro 1.9.2 without any build failure.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -95,6 +95,7 @@
     <ant.version>1.9.1</ant.version>
     <antlr.version>3.4</antlr.version>
     <avro.version>1.7.5</avro.version>
+    <avroHelper.version>0.2.48</avroHelper.version>
```
Comment (srramach) [inline pom.xml]: Can you use a more recent version? 0.2.71 is the latest release.

**Comment by srramach** [comment]
> 3\. With these changes gobblin build failures coming from hive serde library is resolved

To be clear, you are saying that with this snapshot, Gobblin is able to use Avro 1.9.2, without any build failures, right?

File: pom.xml (hunk 1 above)
Comment (abhisheknath2011) [reply pom.xml]: Added latest avro-util version 0.2.71.

**Comment by abhisheknath2011** [comment]
> > 3. With these changes gobblin build failures coming from hive serde library is resolved
> 
> To be clear, you are saying that with this snapshot, Gobblin is able to use Avro 1.9.2, without any build failures, right? 
Yes. I have updated the description with this.

**Review by srramach — APPROVED** [review]

**Pull request merged**